command/agent/internal/ctmanager/runner_config.go

```diff
@@ -0,0 +1,149 @@
+package ctmanager
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	ctconfig "github.com/hashicorp/consul-template/config"
+	ctlogging "github.com/hashicorp/consul-template/logging"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/vault/command/agent/config"
+	"github.com/hashicorp/vault/sdk/helper/pointerutil"
+)
+
+type ManagerConfig struct {
+	AgentConfig *config.Config
+	Namespace   string
+	LogLevel    hclog.Level
+	LogWriter   io.Writer
+}
+
+// NewManagerConfig returns a consul-template runner configuration, setting the
+// Vault and Consul configurations based on the clients configs.
+func NewManagerConfig(sc ManagerConfig, templates ctconfig.TemplateConfigs) (*ctconfig.Config, error) {
```

Comment:
The naming here is a bit strange (creating a NewManagerConfig from a ManagerConfig),

Perhaps keep it as NewRunnerConfig?

Reply:
> Also tiny thing, sc should probably be renamed to mc.

updated